frontend/public/components/modals/cluster-update-modal.tsx

```diff
@@ -36,14 +51,46 @@ const ClusterUpdateModal = withHandlePromise((props: ClusterUpdateModalProps) =>
       ? currentMinorVersionPatchUpdate?.version
       : availableSortedUpdates[0]?.version) || '',
   );
+  const [machineConfigPools, machineConfigPoolsLoaded] = useK8sWatchResource<
+    MachineConfigPoolKind[]
+  >({
+    isList: true,
+    kind: referenceForModel(MachineConfigPoolModel),
+  });
   const [error, setError] = React.useState(errorMessage);
   const [isOpen, setIsOpen] = React.useState(false);
-  const onToggle = () => setIsOpen(!isOpen);
-  const onSelect = (event, selection) => {
+  const [machineConfigPoolsToPause, setMachineConfigPoolsToPause] = React.useState<string[]>([]);
+  const [upgradeType, setUpgradeType] = React.useState<upgradeTypes>(upgradeTypes.Full);
+  React.useEffect(() => {
+    const initialMCPPausedValues = [];
+    machineConfigPools.map((mcp) => {
+      !isMCPMaster(mcp) && isMCPPaused(mcp)
+        ? initialMCPPausedValues.push(mcp.metadata.name)
+        : _.pull(mcp.metadata.name);
```

Comment:
What are you pulling? I'm pretty sure `_.pull` does not do anything by itself. Also `mcp.metadata.name` is a string. Whereas [the docs](https://lodash.com/docs/4.17.15#pull) say it pulls from arrays.

Did you mean to do:
```
_.pull(initialMCPPausedValues, mcp.metadata.name);
```